x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_logic.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { HttpError } from '../../../../../common/types/api';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+
+import { FetchEnginesAPILogic } from '../../api/engines/fetch_engines_api_logic';
+
+import { DEFAULT_META, EngineListDetails, Meta, updateMetaPageIndex } from './types';
+
+export interface EnginesListActions {
+  apiError(error: HttpError): HttpError;
+  apiSuccess({ enginesList, meta }: { enginesList: EngineListDetails[]; meta: Meta }): {
+    enginesList: EngineListDetails[];
+    meta: Meta;
+  };
+  fetchEngines({ meta }: { meta: Meta }): { meta: Meta };
+  makeRequest: typeof FetchEnginesAPILogic.actions.makeRequest;
```

Comment:
nitpick: This type definition doesn't work: the type of this property isn't an action, but an action creator--which is a function called to produce the action. Instead you want the action type, which you can get with something like `Actions<FetchEnginesApiParams, FetchEnginesApiResponse>['makeRequest']`